src/Mod/Sketcher/Gui/Command.cpp

```diff
@@ -978,16 +977,223 @@ void CmdSketcherViewSection::activated(int iMsg)
+{
+    Gui::ActionGroup* pcAction = new Gui::ActionGroup(this, Gui::getMainWindow());
+    pcAction->setDropDownMenu(true);
+    pcAction->setExclusive(false);
+    applyCommandData(this->className(), pcAction);
+
+    QAction* a0 = pcAction->addAction(QString());
+    QAction* a1 = pcAction->addAction(QString());
+    QAction* a2 = pcAction->addAction(QString());
+
+    GridSpaceAction* a3 = new GridSpaceAction(pcAction);
+    pcAction->addAction(a3);
+
+    a0->setCheckable(true);
+    a1->setCheckable(true);
+    a2->setCheckable(true);
+
+    _pcAction = pcAction;
+    languageChange();
+
+    return pcAction;
+}
+
+void CmdSketcherCompGrid::languageChange()
+{
+    Command::languageChange();
+
+    if (!_pcAction)
+        return;
+    Gui::ActionGroup* pcAction = qobject_cast<Gui::ActionGroup*>(_pcAction);
+    QList<QAction*> a = pcAction->actions();
+
+    QAction* a0 = a[0];
+    a0->setText(QApplication::translate("CmdSketcherGridToggle", "Activate Grid"));
+    a0->setToolTip(QApplication::translate("CmdSketcherGridToggle", "Activate Grid"));
+    a0->setStatusTip(a0->toolTip());
+    QAction* a1 = a[1];
+    a1->setText(QApplication::translate("CmdSketcherGridSnap", "Grid Snap"));
+    a1->setToolTip(QApplication::translate("CmdSketcherGridSnap", "New points will snap to the nearest grid line.\nPoints must be set closer than a fifth of the grid size to a grid line to snap."));
+    a1->setStatusTip(a1->toolTip());
+    QAction* a2 = a[2];
+    a2->setText(QApplication::translate("CmdSketcherGridAuto", "Grid Auto Resize"));
+    a2->setToolTip(QApplication::translate("CmdSketcherGridAuto", "Resize grid automatically depending on zoom."));
+    a2->setStatusTip(a2->toolTip());
+}
+
+bool CmdSketcherCompGrid::isActive()
+{
+    //Initialisation of the commands to the sketch specific settings
+    Gui::Document* doc = getActiveGuiDocument();
     if (doc) {
-        // checks if a Sketch Viewprovider is in Edit and is in no special mode
-        SketcherGui::ViewProviderSketch* vp = dynamic_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
-        if (vp /*&& vp->getSketchMode() == ViewProviderSketch::STATUS_NONE*/)
-            return true;
+        auto* sketchView = dynamic_cast<SketcherGui::ViewProviderSketch*>(doc->getInEdit());
+        Gui::ActionGroup* pcAction = qobject_cast<Gui::ActionGroup*>(getAction());
+        if (sketchView && pcAction) {
+            QList<QAction*> a = pcAction->actions();
+
+            bool showGrid = sketchView->ShowGrid.getValue();
+            bool gridSnap = sketchView->GridSnap.getValue();
+            bool gridAuto = sketchView->GridAuto.getValue();
+
+            a[0]->setChecked(showGrid);
+            a[1]->setChecked(gridSnap);
+            a[2]->setChecked(gridAuto);
+
+            //activate/deactivate others
+            a[1]->setEnabled(showGrid);
+            a[2]->setEnabled(showGrid);
+            a[3]->setEnabled(showGrid && !gridAuto);

```

Comment:
Maybe this can be done only if group is active, or only when group is clicked on.

Reply:
This was wmayer proposal to use isActive to be able to initialize the tool correctly. As sketchView is not initialized in 'createAction'.
He said it shouldn't be an issue even it it triggers everytime isActive is called.

Reply:
Alternatively a method could be implemented that is connected with the application's signalInEdit method. That's probably the cleanest way and causes the least overhead.

Reply:
I have nothing against using `isActive`. But you can start by evaluating `isEditModeToolActive()` and update values only if it's `true`

Reply:
I think so. this is proper coding. Just because no one know the future. ;)